<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QxrdExperimentPreferencesDialog</class>
 <widget class="QDialog" name="QxrdExperimentPreferencesDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>874</width>
    <height>471</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Qxrd Preferences</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="1" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QTabWidget" name="m_PreferencesPanes">
     <property name="currentIndex">
      <number>2</number>
     </property>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>Detectors</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="1" column="0">
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>288</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="m_MoveDetectorUp">
             <property name="toolTip">
              <string>Move Detector Up</string>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="qxrdresources.qrc">
               <normaloff>:/images/go-up.png</normaloff>:/images/go-up.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="m_MoveDetectorDown">
             <property name="toolTip">
              <string>Move Detector Down</string>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="qxrdresources.qrc">
               <normaloff>:/images/go-down.png</normaloff>:/images/go-down.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="m_ConfigureDetector">
             <property name="toolTip">
              <string>Configure Detector</string>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="../../source/submodules/qceplib/qceplib-base/qcepresources.qrc">
               <normaloff>:/images/preferences.png</normaloff>:/images/preferences.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="m_RemoveDetector">
             <property name="toolTip">
              <string>Delete Detector(s)</string>
             </property>
             <property name="text">
              <string>Remove Detector</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="m_AddDetector">
             <property name="toolTip">
              <string>Add New Detector</string>
             </property>
             <property name="text">
              <string>Add Detector</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="0" column="0">
          <widget class="QTreeView" name="m_DetectorsList">
           <property name="selectionMode">
            <enum>QAbstractItemView::ExtendedSelection</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Expt and Logging</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="1" column="2">
        <widget class="QPushButton" name="m_CurrentLogFileBrowse">
         <property name="toolTip">
          <string>Select new log file</string>
         </property>
         <property name="text">
          <string>Browse...</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Data dir:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="m_DataDirectory">
         <property name="toolTip">
          <string>Directory where data files are saved 
(relative to experiment dir)</string>
         </property>
         <property name="readOnly">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="m_ExperimentDirectory">
         <property name="toolTip">
          <string>Directory where experiment file is saved
Change only by saving a new experiment</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>log file:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="m_CurrentLogFile">
         <property name="toolTip">
          <string>Log file path (relative to experiment dir)</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Experiment dir:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="m_DataDirectoryBrowse">
         <property name="toolTip">
          <string>Select new data directory</string>
         </property>
         <property name="text">
          <string>Browse...</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0" colspan="3">
        <spacer name="verticalSpacer_6">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>678</width>
           <height>182</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Max log file length (MB):</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QSpinBox" name="m_LogFileMaxMB">
         <property name="maximum">
          <number>10000</number>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Number of log file backups:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QSpinBox" name="m_LogFileBackups"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Image Saving</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_7">
       <item row="4" column="1" colspan="4">
        <widget class="QLineEdit" name="m_SaveDarkSubdir"/>
       </item>
       <item row="2" column="1" colspan="4">
        <widget class="QLineEdit" name="m_SaveRawSubdir"/>
       </item>
       <item row="17" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Width of 'phase' field in saved file names</string>
         </property>
        </widget>
       </item>
       <item row="16" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Width of file index field in saved file names</string>
         </property>
        </widget>
       </item>
       <item row="15" column="1">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>&lt;name&gt;</string>
         </property>
        </widget>
       </item>
       <item row="16" column="1" colspan="4">
        <widget class="QSpinBox" name="m_FileIndexWidth">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>20</number>
         </property>
        </widget>
       </item>
       <item row="7" column="5">
        <widget class="QPushButton" name="m_SaveSubtractedBrowse">
         <property name="text">
          <string>Browse...</string>
         </property>
        </widget>
       </item>
       <item row="19" column="1" colspan="4">
        <widget class="QSpinBox" name="m_FileOverflowWidth">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>20</number>
         </property>
        </widget>
       </item>
       <item row="21" column="0" colspan="6">
        <spacer name="verticalSpacer_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>108</width>
           <height>139</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="15" column="5">
        <widget class="QComboBox" name="m_FileNameFormat4"/>
       </item>
       <item row="18" column="1" colspan="4">
        <widget class="QSpinBox" name="m_FileNumberWidth">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>20</number>
         </property>
        </widget>
       </item>
       <item row="15" column="2">
        <widget class="QComboBox" name="m_FileNameFormat1"/>
       </item>
       <item row="17" column="1" colspan="4">
        <widget class="QSpinBox" name="m_FilePhaseWidth">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>20</number>
         </property>
        </widget>
       </item>
       <item row="4" column="5">
        <widget class="QPushButton" name="m_SaveDarkBrowse">
         <property name="text">
          <string>Browse...</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0">
        <widget class="QCheckBox" name="m_SaveSubtractedInSubdir">
         <property name="text">
          <string>Save Subtracted Data in Subdirectory?</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1" colspan="4">
        <widget class="QLineEdit" name="m_SaveSubtractedSubdir"/>
       </item>
       <item row="20" column="1" colspan="4">
        <widget class="QSpinBox" name="m_DetectorNumberWidth">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>20</number>
         </property>
        </widget>
       </item>
       <item row="15" column="4">
        <widget class="QComboBox" name="m_FileNameFormat3"/>
       </item>
       <item row="2" column="0">
        <widget class="QCheckBox" name="m_SaveRawInSubdir">
         <property name="text">
          <string>Save Raw Data in Subdirectory?</string>
         </property>
        </widget>
       </item>
       <item row="13" column="0">
        <widget class="QCheckBox" name="m_SaveOverflowFiles">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="toolTip">
          <string>Save info about overflowed pixels in .overflow file</string>
         </property>
         <property name="text">
          <string>Save Overflow Pixel Information?</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="20" column="0">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>Width of 'detector number' field in saved file names</string>
         </property>
        </widget>
       </item>
       <item row="2" column="5">
        <widget class="QPushButton" name="m_SaveRawBrowse">
         <property name="text">
          <string>Browse...</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QCheckBox" name="m_SaveDarkInSubdir">
         <property name="text">
          <string>Save Dark Data in Subdirectory?</string>
         </property>
        </widget>
       </item>
       <item row="15" column="3">
        <widget class="QComboBox" name="m_FileNameFormat2"/>
       </item>
       <item row="19" column="0">
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>Width of 'overflow' field in saved file names</string>
         </property>
        </widget>
       </item>
       <item row="15" column="0">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Output file numbering format</string>
         </property>
        </widget>
       </item>
       <item row="18" column="0">
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>Width of 'index' field in saved file names</string>
         </property>
        </widget>
       </item>
       <item row="14" column="0">
        <widget class="QCheckBox" name="m_CompressOutputBZ2">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="toolTip">
          <string>Save info about overflowed pixels in .overflow file</string>
         </property>
         <property name="text">
          <string>Compress Output Files with BZIP2</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_5">
      <attribute name="title">
       <string>Saving</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="6" column="1">
        <widget class="QLineEdit" name="m_AccumulateIntegratedName"/>
       </item>
       <item row="2" column="1">
        <widget class="QLineEdit" name="m_SaveIntegratedSubdir"/>
       </item>
       <item row="4" column="1">
        <widget class="QComboBox" name="m_ScanDataNegative">
         <item>
          <property name="text">
           <string>Save Negative values unchanged</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Save Negative values as zero</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Omit Negative values</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="m_SaveIntegratedBrowse">
         <property name="text">
          <string>Browse...</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QCheckBox" name="m_SaveIntegratedInSeparateFiles">
         <property name="text">
          <string>Save Integrated Data in Individual Files?</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="m_IntegratedScansFileBrowse">
         <property name="text">
          <string>Browse...</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="m_ScanFileExtension"/>
       </item>
       <item row="7" column="0">
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>Max scan file length (MB):</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>For Negative intensities in integrated data</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>File extension for integrated data</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QCheckBox" name="m_SaveIntegratedInLogFile">
         <property name="text">
          <string>Save Integrated Data in Scans File?</string>
         </property>
        </widget>
       </item>
       <item row="9" column="0" colspan="4">
        <spacer name="verticalSpacer_7">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>628</width>
           <height>75</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="0">
        <widget class="QCheckBox" name="m_SaveIntegratedInSubdir">
         <property name="text">
          <string>Save Integrated Data in Subdirectory?</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0" colspan="4">
        <widget class="QCheckBox" name="m_AccumulateIntegrated2D">
         <property name="enabled">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Accumulate Integrated Data into 2-D Dataset?</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="m_IntegratedScansFile">
         <property name="toolTip">
          <string>Integrated Scans File (relative to data directory)</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_17">
         <property name="text">
          <string>Data name for accumulated 2-D Dataset</string>
         </property>
        </widget>
       </item>
       <item row="8" column="0">
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>Number of backup scan files:</string>
         </property>
        </widget>
       </item>
       <item row="7" column="1">
        <widget class="QSpinBox" name="m_ScanFileMaxMB">
         <property name="maximum">
          <number>10000</number>
         </property>
        </widget>
       </item>
       <item row="8" column="1">
        <widget class="QSpinBox" name="m_ScanFileBackups"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Servers</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="0" column="0">
        <widget class="QCheckBox" name="m_RunSpecServer">
         <property name="text">
          <string>Run 'spec' server socket</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>On Port:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="m_SpecServerPort">
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::NoButtons</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <widget class="QCheckBox" name="m_RunSimpleServer">
         <property name="text">
          <string>Run 'simple' server socket</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>On Port:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="m_SimpleServerPort">
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::NoButtons</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="2" column="0">
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>134</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../source/submodules/qceplib/qceplib-base/qcepresources.qrc"/>
  <include location="qxrdresources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>QxrdExperimentPreferencesDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>QxrdExperimentPreferencesDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
